types/node/compatibility/iterators.d.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Backwards-compatible IteratorObject interface, which is augmented with iterator helper methods by lib.esnext.iterator in TypeScript 5.6.
+ * The IterableIterator interface does not contain these methods, which creates assignability issues in places where IteratorObjects
+ * are expected (eg. DOM-compatible APIs) if lib.esnext.iterator is loaded.
+ * Also ensures that iterators returned by the Node API, which inherit from Iterator.prototype, correctly expose the iterator helper methods
+ * if lib.esnext.iterator is loaded.
+ * TODO: remove once this package no longer supports TS 5.5.
+ */
+
+interface IteratorObject<T, TReturn = unknown, TNext = unknown> extends Iterator<T, TReturn, TNext> {
```

Comment:
It might be better to do as `RelativeIndexable` has done to avoid potential lib conflicts. It's perhaps possible to implement a forward-declaration mechanism that is safer:

```ts
// global.d.ts

// NodeJS-specific declaration, pulls in forward-declared types
interface NodeIteratorObject<T, TReturn = unknown, TNext = unknown> extends Iterator<T, TReturn, TNext> {
  [Symbol.iterator](): NodeIteratorObject<T, TReturn, TNext>;
}

// iterators.d.ts

// forward declaration
interface IteratorObject<T, TReturn, TNext> {} 

// NodeJS-specific declaration, pull in forward declaration
interface NodeIteratorObject<T, TReturn, TNext> extends IteratorObject<T, TReturn, TNext> {
}
```